core/roboconf-dm-rest-services/src/main/java/net/roboconf/dm/rest/services/internal/resources/DebugResource.java

```diff
@@ -155,8 +159,51 @@ else if( ma.getApplication().getGraphs().getRootComponents().size() == 1
 	 * #checkMessagingConnectionForTheDm()
 	 */
 	@Override
-	public Response checkMessagingConnectionForTheDm() {
-		return Response.status( Status.SERVICE_UNAVAILABLE ).entity( "Not yet implemented." ).build();
+	public Response checkMessagingConnectionForTheDm( String message, long timeout ) {
+
+		if ( message == null ) {
+			// Generates an unique message.
+			message = "ECHO " + UUID.randomUUID();
+		}
+
+		// Max timeout
+		if ( timeout > MAXIMUM_TIMEOUT ) {
+			this.logger.warning( "Timeout " + timeout + "ms is above maximum limit " + MAXIMUM_TIMEOUT + "ms. Normalizing" );
+			timeout = MAXIMUM_TIMEOUT;
+		}
+
+		this.logger.fine( "Checking connection to the message queue. message=" + message
+				+ ", timeout=" + timeout + "ms" );
+
+		boolean hasReceived;
+		try {
+			// Send the Echo request!
+			hasReceived = this.manager.pingMessageQueue( message, timeout );
+
+		} catch ( IOException e ) {
+			// Something bad has happened!
+			final String eMessage = "Unable to send Echo message " + message;
+			Utils.logException( this.logger, Level.SEVERE, new RuntimeException( eMessage, e ) );
```

Comment:
That's my fault, I edited my comment yesterday but it was probably not propagated by e-mail.
The message must be logged in SEVERE mode and the stack trace in FINEST. So, your code should look like...

```
this.logger.severe( eMessage );
Utils.logException( this.logger, e );
```